### DISCUSSION TITLE
AND operator in if: condition

### DISCUSSION THREAD

**Discussion by reikdas** [community/community#27152] [Actions]
<p>How do I use an <code>&amp;&amp;</code> operator in the <code>if:</code> block of a step to check for multiple conditions?</p>
<p>For eg -<br>
<code>if: (${{ matrix.coverage }} == true) &amp;&amp; (runner.os == 'Linux')</code><br>
doesn’t work.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>You need to enclose the whole expression in <code>${{ ... }}</code>, or nothing at all:</p>
<pre><code class="lang-yaml">if: ${{ (matrix.coverage == true) &amp;&amp; (runner.os == 'Linux') }}
</code></pre>

**Reply by socketbox** [reply]
How is this not a bug, given what's indicated in the documentation regarding `if` and the evaluation of expression syntax? To wit:
>When you use expressions in an if conditional, you may omit the expression syntax (${{ }}) because GitHub automatically evaluates the if conditional as an expression. For more information about if conditionals, see "[Workflow syntax for GitHub Actions](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idif)."
https://docs.github.com/en/actions/learn-github-actions/expressions#about-expressions

**Reply by airtower-luna** [reply]
No `${{ ... }}` at all should work, too. But using it for *parts* of the `if` value will always create a non-empty string, which is always `true`.

**Reply by socketbox** [reply]
Let me just say, before continuing, that I'm very grateful to you for sharing this knowledge.

These combinations did not work for me:
`if: ${{ success() }} && ${{ matrix.category == 'foo' }}`
and, as you've mentioned, any combination that involves enclosing one clause in `${{ }}`, but not the other.  I haven't tested an expression that does not group clauses with `( )` (ie. `${{ success() && matrix.category == 'foo' }}`

I think either the parser needs to be improved or I should edit the docs, unless you know of a place in the documentation that states explicitly the rules that you've shared.

EDIT:
My initial response was incorrect: `if: success() && matrix.category == 'foo'` does work.
